File: ios/Messages.swift
import Foundation
import ExpoModulesCore
import LibSignalClient

final class SignalMessageRef: SharedObject {
  let message: SignalMessage

  init(message: SignalMessage) {
    self.message = message
    super.init()
  }
}

final class PreKeySignalMessageRef: SharedObject {
  let message: PreKeySignalMessage

  init(message: PreKeySignalMessage) {
    self.message = message
    super.init()
  }
}

File: android/src/main/java/expo/modules/libsignal/Messages.kt
package expo.modules.libsignal

import expo.modules.kotlin.sharedobjects.SharedObject
import org.signal.libsignal.protocol.message.PreKeySignalMessage
import org.signal.libsignal.protocol.message.SignalMessage

class SignalMessageRef(val message: SignalMessage) : SharedObject()

class PreKeySignalMessageRef(val message: PreKeySignalMessage) : SharedObject()
